src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(MakeLocalFunctionStaticCodeRefactoringProvider)), Shared]
+    internal sealed class MakeLocalFunctionStaticCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
```

Comment:
>ComputeRefactoringsAsync [](start = 35, length = 24)

I believe there is a case for making this an analyzer/fixer as opposed to a refactoring. It seems very reasonable for a code base to want to ban use of non-static local functions as it there is a trade off between verbose call sites/parameter lists for static ones versus more hard to understand code from implicit captures and state updates for non-static ones. It seems like all the code prior to registering the refactoring can easily be moved to an analyzer.

Reply:
Makes sense. Probably not gonna change it in this PR, but I will create an issue for this.